src/radical/pilot/pmgr/launching/base.py

```diff
@@ -801,6 +801,11 @@ def _prepare_pilot(self, resource, rcfg, pilot, expand, tar_name):
 
             requested_nodes = math.ceil(requested_nodes)
 
+        # now that we know the number of nodes to request, calculate back the
+        # actual number of requested cores and gpus
+        requested_cores = requested_nodes * cores_per_node
+        requested_gpus  = requested_nodes * gpus_per_node
```

Comment:
should be removed, since duplicates lines above (which also uses `avail_(cores|gpus)_per_node`)

Reply:
I see - well, it should be indented then I guess, because w/o that code the pilot will report incorrect amount of resources and analytics will be thrown off (percentages don't work out).

Reply:
The code now makes a difference between `requested_cores/gpus` and `allocated_cores/gpus`, and the latter is what is reported as pilot resources for analytics.